### PR TITLE
Ext transform feedback

### PR DESCRIPTION
**Pull request by yalcinmelihyasin**
Implement VK_EXT_transform_feedback
b/192669426

### CODE REVIEW

**Review by matled — COMMENTED** [review]
Thanks!

I don't think I can review this properly yet, found just one stylistic nit :)

File: gapis/api/vulkan/command_buffer_rebuilder.go [hunk 1]
```diff
@@ -1416,6 +1416,155 @@ func rebuildVkCmdDispatchBase(
...
+			offsetData.Free()
+			sizesData.Free()
+		},
+		cb.VkCmdBindTransformFeedbackBuffersEXT(commandBuffer,
+			d.FirstBinding(),
+			d.BindingCount(),
+			bufferData.Ptr(),
+			offsetData.Ptr(),
+			sizesData.Ptr(),
+		).AddRead(offsetData.Data()).AddRead(bufferData.Data()).AddRead(sizesData.Data()),
```
Comment (pmuetschard) [inline gapis/api/vulkan/command_buffer_rebuilder.go:1449]: nit: style-wise, I'd prefer each `.AddRead` to be on its own line.

**Review by pmuetschard — COMMENTED** [review]
I've taken an initial look at this and so far everything seems fine. I do have to spend a bit more time with it though in order to understand everything.

File: gapis/api/vulkan/extensions/ext_transform_feedback.api [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+        readMemoryInCurrentPipelineBoundVertexBuffers(0xFFFFFFFF, 0xFFFFFFFF, 0, 0)
+        clearLastDrawInfoDrawCommandParameters()
+        ldi := lastDrawInfo()
+        ldi.CommandParameters.DrawIndirectByteCountEXT = args
+    }
+}
+
+@extension("VK_EXT_transform_feedback")
+@indirect("VkCommandBuffer", "VkDevice")
+cmd void vkCmdDrawIndirectByteCountEXT(
```
Comment (pmuetschard) [inline gapis/api/vulkan/extensions/ext_transform_feedback.api:427]: does this need `@executed_draw`?

Comment (yalcinmelihyasin) [reply gapis/api/vulkan/extensions/ext_transform_feedback.api:427]: correct

File: gapis/api/vulkan/command_buffer_rebuilder.go [hunk 3]
```diff
@@ -1416,6 +1416,161 @@ func rebuildVkCmdDispatchBase(
...
+			offsetData.Free()
+			sizesData.Free()
+		},
+		cb.VkCmdBindTransformFeedbackBuffersEXT(commandBuffer,
+			d.FirstBinding(),
+			d.BindingCount(),
+			bufferData.Ptr(),
+			offsetData.Ptr(),
+			sizesData.Ptr(),
+		).AddRead(
```
Comment (pmuetschard) [inline gapis/api/vulkan/command_buffer_rebuilder.go:1449]: Thanks for the change. Please also do it for the other ones in this file.

**Review by pmuetschard — APPROVED** [review]

**Pull request merged**